src/struct.c

```diff
@@ -631,6 +631,52 @@ Dsymbol *AggregateDeclaration::searchCtor()
     return s;
 }
 
+/***************************************
+ * Return true if aggregate is an InputRange.
+ */
+bool AggregateDeclaration::isInputRange()
+{
+    return (this->search(Loc(), Id::Ffront) &&
```

Comment:
Should I implement checks for this?

I'm guessing I should use `Scope::search()` when searching for free-standing property functions right?

For every such free function `f` how I do check that it's callable as `f(e)`?